### DISCUSSION TITLE
[suggestion] Better windows

### DISCUSSION THREAD

**Discussion by elro0003** [tsunamayo/Starship-EVO#3969] [Suggestions]
At the moment the windows are pretty much the only limiting factor in the shape and size of cockpits and ships. Any windows with any curvature or bigger that 4m will not look good because of how the current window tile wedge, window arch, and window tile tetra work.

Example:

![IMG_20210501_084753](https://user-images.githubusercontent.com/69245083/116772794-f0946380-aa59-11eb-9452-6c30a9602be7.jpg)

So the things that need to be fixed are:
- The size of the windows (down to 0.25), and when the scale update comes the possibility to scale them down to 0.125m and also to the max size
- the way the tetras join together

I have 2 ideas on how to fix these issues.

The first one is to add a new block called the inverse tetra or just hepta. A window that essentially is a hepta block to windows, and smoothly join with the tetra windows

The second and in my opinion the better idea, is to take note from space engineers. Make / add windows that are alot thinner. If the windows are just planes, like the ones in space engineers. You would need just 4 new window blocks, or change the shape of the current window blocks:

-Window tetra (just the outer surface of current tetra windows)
-Window panel (just a flat window)
-Window wedge/arch (the outer surface of current window wedges)
And window corner/triangle (a flat triangle)
(None of the windows should have borders, they can easily be added with bricks)
And a 5th window that is the corner bricks tilted triangle surface would make the possibilities even more endless

This way you could also have the windows be a smaller scale than they currently are, and then the windows no longer would limit the builds. I already have quite a few ships where I have just placed black bricks instead of the windows, just because you can't do very curvy or good looking large windows. It'd be great if you get better windows to the game soon.

Edit: 
So with the new scale thing the windows are much better, but it still can be improved a lot
![IMG_20210501_113906](https://user-images.githubusercontent.com/69245083/116776617-f39b4e00-aa71-11eb-9ab5-f3d3b4f2cc96.jpg)

**Comment by Rusty-uy** [comment]
i totally agree, thats true

**Comment by VivaciousVii** [comment]
culling the backface of individual blocks that are not facing the camera could buff fps -and- make windows look more transparent

**Comment by najlitarvan** [comment]
also glass tiles would be nice
